services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/UploadWithUnknownContentLengthHelper.java

```diff
@@ -128,6 +132,9 @@ private class UnknownContentLengthAsyncRequestBodySubscriber implements Subscrib
             this.maximumChunkSizeInByte = maximumChunkSizeInByte;
             this.putObjectRequest = putObjectRequest;
             this.returnFuture = returnFuture;
+            this.progressListener = putObjectRequest.overrideConfiguration()
+                                                    .map(c -> c.executionAttributes().getAttribute(JAVA_PROGRESS_LISTENER))
+                                                    .orElse(null);
```

Comment:
Can we make it no-op listener if it's null so that we don't need to check null downstream?